django/contrib/auth/management/commands/createsuperuser.py

```diff
@@ -138,6 +139,13 @@ def handle(self, *args, **options):
                     user_data[PASSWORD_FIELD] = password
             else:
                 # Non-interactive mode.
+                # Use password from environment variable, if provided.
+                if PASSWORD_FIELD in user_data and 'DJANGO_SUPERUSER_PASSWORD' in os.environ:
+                    user_data[PASSWORD_FIELD] = os.environ['DJANGO_SUPERUSER_PASSWORD']
+                # Use username from environment variable, if not provided in
+                # options.
+                if username is None:
+                    username = os.environ.get('DJANGO_SUPERUSER_' + self.UserModel.USERNAME_FIELD.upper())
```

Comment:
If this statement is executed, that means that username was None at the time. In the `if` statement that follows, it raises a CommandError `if username is None`. Can you help me understand why we're checking for `if username is None` twice, while executing two different statements?

Reply:
First `if username is None` is for fetching the username from `environment variable`. so if it can not be fetched from `environment variable`, the username will be None again and the second `if username is None`.

Reply:
Ahh, I see! Thank you for the quick reply! Learning a lot from these PRs! :)